### PR TITLE
added portal theme support

### LINKED ISSUE #43: Portal theme support
As a user I would like to have theme support for different instance of management portal

### PR DESCRIPTION
**Pull request by Chi-Chu319**
resolves #43

### CODE REVIEW

File: swagger.yaml [hunk 1]
```diff
@@ -1931,6 +1932,222 @@ paths:
...
+  "/v1/portalThemes":
+    get:
+      operationId: listPortalTheme
+      summary: Lists portal themes
+      description: Lists portal themes
+      tags:
+        - PortalTheme
+      responses:
+        "200":
+          description: A list of portal files
```
Comment (Chi-Chu319) [inline swagger.yaml]:
```suggestion
          description: A list of portal themes
```

File: swagger.yaml [hunk 2]
```diff
@@ -1931,6 +1932,222 @@ paths:
...
+          in: path
+          description: portal theme id
+          schema:
+            type: string
+            format: uuid
+      requestBody:
+        $ref: "#/components/requestBodies/PortalTheme"
+      responses:
+        "200":
+          description: an updated export theme file
```
Comment (Chi-Chu319) [inline swagger.yaml]:
```suggestion
          description: an updated portal theme
```

File: swagger.yaml [hunk 3]
```diff
@@ -1931,6 +1932,222 @@ paths:
...
+              schema:
+                $ref: "#/components/schemas/BadRequest"
+        "403":
+          description: Attempted to make a call with unauthorized client
+          content:
+            application/json;charset=utf-8:
+              schema:
+                $ref: "#/components/schemas/Forbidden"
+        "404":
+          description: Attempted to make a call to an non existant metaform
```
Comment (Chi-Chu319) [inline swagger.yaml]:
```suggestion
          description: Attempted to make a call to an non existant theme
```

File: swagger.yaml [hunk 4]
```diff
@@ -1931,6 +1932,222 @@ paths:
...
```
Comment (Chi-Chu319) [inline swagger.yaml]:
```suggestion
          description: Attempted to make a call to an non existant theme
```

File: swagger.yaml [hunk 5]
```diff
@@ -1931,6 +1932,222 @@ paths:
...
```
Comment (Chi-Chu319) [inline swagger.yaml]:
```suggestion
          description: Attempted to make a call to an non existant theme
```

File: swagger.yaml [hunk 6]
```diff
@@ -1931,6 +1932,222 @@ paths:
...
```
Comment (Chi-Chu319) [inline swagger.yaml]:
```suggestion
          description: Attempted to make a call to an non existant theme
```

File: swagger.yaml [hunk 7]
```diff
@@ -1931,6 +1932,222 @@ paths:
...
+            format: uuid
+      requestBody:
+        $ref: "#/components/requestBodies/AdminTheme"
+      responses:
+        "200":
+          description: an updated admin theme
+          content:
+            application/json;charset=utf-8:
+              schema:
+                $ref: "#/components/schemas/ExportThemeFile"
```
Comment (Chi-Chu319) [inline swagger.yaml]:
```suggestion
      responses:
        "200":
          description: an updated admin theme
          content:
            application/json;charset=utf-8:
              schema:
                $ref: "#/components/schemas/AdminTheme"
```

File: swagger.yaml [hunk 8]
```diff
@@ -1931,6 +1932,222 @@ paths:
...
+      operationId: createAdminTheme
+      summary: Create new admin theme
+      description: Creates a new admin theme
+      tags:
+        - AdminTheme
+      requestBody:
+        $ref: "#/components/requestBodies/AdminTheme"
+      responses:
+        "200":
+          description: Created export admin theme
```
Comment (Chi-Chu319) [inline swagger.yaml]:
```suggestion
          description: Created admin theme
```

File: swagger.yaml [hunk 9]
```diff
@@ -2193,6 +2416,42 @@ components:
...
+      required:
+        - data
+      properties:
+        id:
+          type: string
+          format: uuid
+          readOnly: true
+        data:
+          type: object
+        name:
```
Comment (anttileppa) [inline swagger.yaml:2434]: name and slug should be required

**Review by anttileppa — CHANGES_REQUESTED** [review]

**Review by anttileppa — APPROVED** [review]

File: swagger.yaml [hunk 10]
```diff
@@ -2429,7 +2429,7 @@ components:
           format: uuid
           readOnly: true
         data:
-          type: object
+          type: string
```
Comment (Chi-Chu319) [inline swagger.yaml]: The data should be object

File: swagger.yaml [hunk 11]
```diff
@@ -2193,6 +2417,44 @@ components:
               items:
                 $ref: "#/components/schemas/MetaformScript"
 
+    AdminTheme:
+      type: object
+      required:
+        - data
+        - name
+        - slug
```
Comment (Chi-Chu319) [inline swagger.yaml]:
```suggestion
```

**Pull request merged**